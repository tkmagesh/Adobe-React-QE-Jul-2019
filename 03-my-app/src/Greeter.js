import React from 'react';

class Greeter extends React.Component{
	
	state = { 
		message : '[Dummy message]',
		userName : ''
	};

	onBtnGreetClick = () => {
		this.setState({ message : `Hi ${this.state.userName}, ${this.props.msg}`});
	}

	onTxtUserNameChange = (evt) => {
		this.setState({ userName : evt.target.value });
	}

	render(){
		
		return(
			<div>
				<h1>Greeter</h1>
				<hr/>
				<br/>
				<label htmlFor="">User Name :</label>
				<input type="text" onChange={this.onTxtUserNameChange} />
				<input type="button" value="Greet" onClick={this.onBtnGreetClick} />
				<div className="highlight">{this.state.message}</div>
			</div>
		)
	}
}

export default Greeter;