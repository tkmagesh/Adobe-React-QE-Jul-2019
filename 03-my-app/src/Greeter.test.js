import React from 'react';
import Greeter from './Greeter';
import { shallow } from 'enzyme';

describe('Greeter Component', () => {
	test('should match the snapshot', () => {
		let greeter = shallow(<Greeter msg="Have a nice day"/>);

		expect(greeter).toMatchSnapshot();
	});

	test('should display the greet message', () => {
		let greeter = shallow(<Greeter msg="Have a nice day"/>);

		greeter.setState({userName : 'Magesh'});
		greeter
			.find('input[value="Greet"]')
			.simulate('click');

		expect(greeter
			.find('.highlight')
			.text()
		)
		.toBe('Hi Magesh, Have a nice day');

		expect(greeter).toMatchSnapshot();
	});

	test('should update the state with the greet message', () => {
		let greeter = shallow(<Greeter msg="Have a nice day"/>);

		greeter.setState({userName : 'Magesh'});
		greeter
			.find('input[value="Greet"]')
			.simulate('click');

		expect(greeter.state('message'))
			.toBe('Hi Magesh, Have a nice day');

		expect(greeter).toMatchSnapshot();
	});

})