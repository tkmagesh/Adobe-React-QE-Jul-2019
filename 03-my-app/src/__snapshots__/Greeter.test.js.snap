// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter Component should display the greet message 1`] = `
<div>
  <h1>
    Greeter
  </h1>
  <hr />
  <br />
  <label
    htmlFor=""
  >
    User Name :
  </label>
  <input
    onChange={[Function]}
    type="text"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Greet"
  />
  <div
    className="highlight"
  >
    Hi Magesh, Have a nice day
  </div>
</div>
`;

exports[`Greeter Component should match the snapshot 1`] = `
<div>
  <h1>
    Greeter
  </h1>
  <hr />
  <br />
  <label
    htmlFor=""
  >
    User Name :
  </label>
  <input
    onChange={[Function]}
    type="text"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Greet"
  />
  <div
    className="highlight"
  >
    [Dummy message]
  </div>
</div>
`;

exports[`Greeter Component should update the state with the greet message 1`] = `
<div>
  <h1>
    Greeter
  </h1>
  <hr />
  <br />
  <label
    htmlFor=""
  >
    User Name :
  </label>
  <input
    onChange={[Function]}
    type="text"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Greet"
  />
  <div
    className="highlight"
  >
    Hi Magesh, Have a nice day
  </div>
</div>
`;
